packages/core-bridge/index.d.ts

```diff
@@ -348,10 +356,19 @@ export interface Client {
 export interface EphemeralServer {
   type: 'EphemeralServer';
 }
+export interface HistoryPusher {
+  type: 'HistoryPusher';
+}
+export interface ReplayWorker {
+  type: 'ReplayWorker';
+  worker: Worker;
+  pusher: HistoryPusher;
```

Comment:
Not worth blocking this PR but I feel like this approach is needlessly adding complexity to every lang SDK.

Reply:
No, because then there need to be different worker types. There is no replay worker at the moment, which is nice. Initially I did have two different types and that made other things harder. We can go that direction if more divergent needs crop up.

The amount of complexity it adds is I think pretty minor. It's even easier in pyo3. IMO Sharing a `feed_history` implementation for sync task dispatch wouldn't make much sense because the semantics are entirely different for replay vs. pushing new history.

Reply:
I too feel that there is some unjustified complexity added in this PR, mostly in worker.ts and runtime.ts.

I think what really bothers me is the fact that, in Worker.ts, regular workers and replay workers follow distinct initialization/registration paths, making it much less obvious who owns which resource at which moment, and the fact that we have that `historyPublisher` object that needs to be passed over, in parallel with the corresponding worker instance.

I'd say however that this can be fixed with some small refactoring on the TS side only, without impacting core-bridge. Not blocking IMHO.

Reply:
Not sure what you're suggesting @mjameswh but I'm open to hear it, it's not too late.